GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -239,15 +239,18 @@ public String getName() {
     }
 
     /**
-     * Set the (optional) name of the object.
+     * Set the (optional) name of the object. Scene object names are not needed:
+     * they are only for the application's convenience.
      * 
-     * Scene object names are not needed: they are only for the application's
-     * convenience.
-     * 
-     * @param name
-     *            Name of the object.
+     * @param name Name of the object.
+     * @throws IllegalArgumentException if input parameter is null or empty.
      */
     public void setName(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
Yes, we must check the given string for white spaces by suppressing them all, either leading or trailing. The trimmed string only can be used as a valid key in the map. Unfortunately, as I pointed out previously, Preconditions.checkStringNotNullOrEmpty() is not suitable for this case, as it just check null and strlen zero. Considering usage of this setName(), performance is not a problem.